### PR TITLE
Camera QoL Updates + Minor Bug Fix

### PR DESCRIPTION
**Pull request by sololegends**
- Added visible point of rotation for the Camera tilt
- Added switch, `Camera.pivotOnCenter`, to alter tilt behaviour
  - TRUE[default]: Behaviour the typical, expected behaviour.
		Rotate origin is screen space center ray cast to globe
  - FALSE: Rotate origin is mouse point of tilt operation start
- [3D Only] When zooming out, with the camera not facing globe center point,
	the camera will slowly rotate itself to be face the globe center point
	- This also fixes a bug in the camera.
		When zooming out with mouse point off the globe the globe can move out of view entirely

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @sololegends! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/CesiumGS/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/sololegends/cesium/blob/sololegends-camera-updates/CONTRIBUTORS.md) file!
* :x: Missing CLA.
    * Please send in a [Contributor License Agreement](https://github.com/CesiumGS/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) and comment back here to let us know to check this!
    * Maintainers, [here's how to check](https://github.com/CesiumGS/cesium/blob/master/Documentation/Contributors/CLAs/README.md) for a submitted CLA.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/sololegends-camera-updates/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by sololegends** [comment]
Sent CLA, updated `CONTRIBUTORS.md` and `CHANGES.md` as bot requested

**Comment by ebogo1** [comment]
Thanks for the PR @sololegends! I'll take a look at this some this week.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @sololegends! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

File: Source/Scene/Camera.js [hunk 1]
```diff
@@ -338,6 +343,67 @@ function updateCameraDeltas(camera) {
   }
 }
 
+/**
+ * Initialize the camera pivot point in the DOM
+ */
+Camera.prototype._initPivotElement = function () {
```
Comment (sanjeetsuhag) [inline Source/Scene/Camera.js:349]: We prefer to not to prefix functions in the public API with underscores.

File: Source/Scene/Camera.js [hunk 2]
```diff
@@ -338,6 +343,67 @@ function updateCameraDeltas(camera) {
...
+  var camera = this;
+
+  // Initialize the camera pivot element, if needed
+  if (document.getElementById("cesium_camera_pivot") === null) {
+    camera._domPivot = document.createElement("div");
+    camera._domPivot.setAttribute(
```
Comment (sanjeetsuhag) [inline Source/Scene/Camera.js:355]: Ideally, the style of this DOM element should be something the user can configure.

File: Source/Scene/ScreenSpaceCameraController.js [hunk 3]
```diff
@@ -934,9 +955,39 @@ function handleZoom(
...
+  // Negative numbers for zooming out, positive for zooming in
+  if (mode === SceneMode.SCENE3D && distance < 0) {
+    zoomPitchDown();
+  }
   if (!object._cameraUnderground) {
     camera.setView(scratchZoomViewOptions);
   }
+
+  // Pitch down during zoom operation
+  function zoomPitchDown() {
```
Comment (sanjeetsuhag) [inline Source/Scene/ScreenSpaceCameraController.js:967]: This should be declared as a module scoped function.

File: Source/Scene/ScreenSpaceCameraController.js [hunk 4]
```diff
@@ -2291,7 +2342,20 @@ function tilt3DOnEllipsoid(controller, startPosition, movement) {
   var windowPosition = tilt3DWindowPos;
   windowPosition.x = canvas.clientWidth / 2;
   windowPosition.y = canvas.clientHeight / 2;
-  var ray = camera.getPickRay(windowPosition, tilt3DRay);
+  var ray;
+  // Sets the rotational pivot point to the window center
+  //   Else, the rotational pivot point to the drag start
```
Comment (sanjeetsuhag) [inline Source/Scene/ScreenSpaceCameraController.js:2347]:
```suggestion
  // Else, the rotational pivot point to the drag start
```

File: Source/Scene/ScreenSpaceCameraController.js [hunk 5]
```diff
@@ -2291,7 +2342,20 @@ function tilt3DOnEllipsoid(controller, startPosition, movement) {
...
+  if (camera.pivotOnCenter) {
+    ray = camera.getPickRay(windowPosition, tilt3DRay);
+  } else if (startPosition) {
+    ray = camera.getPickRay(startPosition, tilt3DRay);
+  } else {
+    var normalPos = Cartesian3.normalize(
```
Comment (sanjeetsuhag) [inline Source/Scene/ScreenSpaceCameraController.js:2353]: Scratch variables should be utilized here.

**Review by sanjeetsuhag — COMMENTED** [review]
Thanks for your contribution @sololegends. I do like the idea of adding a DOM element to indicate the point of rotation. However, I did notice some weird behavior when the camera is tilted up from the surface of the Earth.

camera2

Aside from that, I have left a few comments regarding the implementation, specifically regarding what the public API for this feature should look like. Additionally, I would recommend adding some more tests to ensure we have good coverage for this functionality and that it works in all edge cases (far away, close to surface, tilted etc).

**Comment by ggetz** [comment]
@sololegends We're trying to be more responsive to PRs and keep our PR backlog tidy. Due to inactivity, I'm going to close this PR. But please feel free to re-open when you get the chance to work on this! Thanks for your patience.
